src/utils/hoc.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+import _ from 'lodash/fp'
+import StripedLoader from './StripedLoader'
+
+export let withNode = (Component, { allowEmptyNode = false } = {}) => ({
+  tree,
+  node,
+  path,
+  ...props
```

Comment:
Since we're passing along all of the props including tree, node, and path, it might be cleaner to just call this props, destructure inside the function, and just pass props along

Something like this:

```js
export let withNode = (Component, { allowEmptyNode = false } = {}) => props => {
  let { tree, node, path } = props
  props.node = node || (tree && path && tree.getNode(path))

  if (!props.node && !allowEmptyNode)
    throw Error(`Node not provided, and couldn't find node at ${path}`)

  return <Component {...props} />
}
```